test/recipeModuleManager.test.js

```diff
@@ -45,4 +57,674 @@ describe(`recipeModuleManagerTest: ${printPath("[test/recipeModuleManager.test.j
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res);
+                    }
+                })
+        );
+        assert(r1.text === "success TestRecipe /hello" || r1.text === "success TestRecipe1 /hello");
+
+        r1 = await new Promise((resolve) =>
+            request(app)
+                .post("/auth/hello")
+                .set("rid", "testRecipe1")
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res);
+                    }
+                })
+        );
+        assert(r1.text === "success TestRecipe1 /hello");
+    });
+
+    //  TODO: Test various inputs to errorHandler (if it accepts or not)
+    it("test various inputs to errorHandler", async function () {
+        await startST();
+
+        ST.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [TestRecipe.init(), TestRecipe1.init()],
+        });
+
+        const app = express();
+
+        app.use(ST.middleware());
+        app.use(ST.errorHandler());
+        app.use((err, request, response, next) => {
+            if (err.message == "General error from TestRecipe") {
+                response.status(200).send("General error handled in user error handler");
+            } else {
+                response.status(500).send("Invalid error");
+            }
+        });
+
+        let r1 = await new Promise((resolve) =>
+            request(app)
+                .post("/auth/error/general")
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res.text);
+                    }
+                })
+        );
+        assert(r1 === "General error handled in user error handler");
+
+        r1 = await new Promise((resolve) =>
+            request(app)
+                .post("/auth/error/badinput")
+                .expect(400)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res);
+                    }
+                })
+        );
+        assert(r1.status === 400);
+        assert(r1.body.message === "Bad input error from TestRecipe");
+    });
+
+    // TODO: Error thrown from APIs implemented by recipes must not go unhandled
+    it("Error thrown from APIs implemented by recipes must not go unhandled", async function () {
```

Comment:
You are testing this API with a non-general error.. also call an API that throws a `new Error("...")` and make sure that your error handler (which you need to define) is called with that error. If there is no such API in the TestRecipe, create one.